### ISSUE TITLE
esbuild binary not found when using pnpm

### ISSUE THREAD

**Issue by zaiste** [hashicorp/next-mdx-remote#129]
When using `pnpm` the `esbuild` binary cannot be located, i.e.

```
wait  - compiling...
event - compiled successfully
node:events:346
      throw er; // Unhandled 'error' event
      ^

Error: spawn /<my project>/node_modules/esbuild/bin/esbuild ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:282:19)
    at onErrorNT (node:internal/child_process:480:16)
    at processTicksAndRejections (node:internal/process/task_queues:81:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (node:internal/child_process:288:12)
    at onErrorNT (node:internal/child_process:480:16)
    at processTicksAndRejections (node:internal/process/task_queues:81:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn /<my project>/node_modules/esbuild/bin/esbuild',
  path: '/<my project>/node_modules/esbuild/bin/esbuild',
  spawnargs: [ '--service=0.11.18', '--ping' ]
}
 ERROR  Command failed with exit code 1.
```

Adding `esbuild` directly to project's `package.json` solves the problem.

**Comment by BRKalow** [comment]
Thanks for the report! We're talking about ways to improve the esbuild binary detection amongst the team.

Would you be able to share a path to the binary without adding `esbuild` directly? That would be helpful!

Thanks!

**Comment by zaiste** [comment]
Thanks for your work on this fantastic package!

Right now the binary could be found at `/node_modules/.pnpm/esbuild@0.11.18/node_modules/esbuild/bin`, but there is a related discussion over at PNPM (cf. https://github.com/pnpm/pnpm/discussions/3191) to provide a more general support. It's not a big issue, this could way till pnpm supports it.

**Comment by kylegach** [comment]
I was able to workaround this by following related guidance from an alternative to this project: https://github.com/kentcdodds/mdx-bundler#nextjs-esbuild-enoent.

TL;DR: Add something like this before you call `serialize`:

```js
if (process.platform === 'win32') {
  process.env.ESBUILD_BINARY_PATH = path.join(
    process.cwd(),
    'node_modules',
    'esbuild',
    'esbuild.exe'
  );
} else {
  process.env.ESBUILD_BINARY_PATH = path.join(
    process.cwd(),
    'node_modules',
    'esbuild',
    'bin',
    'esbuild'
  );
}
```

**Comment by BRKalow** [comment]
@kylegach We actually do have something like that baked in 😅  https://github.com/hashicorp/next-mdx-remote/blob/main/src/serialize.ts#L11-L38

**Comment by shannonrothe** [comment]
I'm facing the same issue here with the following error running Next with `yarn dev`:

```
events.js:292
      throw er; // Unhandled 'error' event
      ^

Error: spawn /Users/shannonrothe/projects/platypus-ui/packages/platypus-docs/node_modules/esbuild/bin/esbuild ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:269:19)
    at onErrorNT (internal/child_process.js:465:16)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
    at onErrorNT (internal/child_process.js:465:16)
    at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  errno: -2,
```

Is there a workaround for this? 😄

**Comment by BRKalow** [comment]
@zaiste Could you try to upgrade to the latest version and see if that resolves your issue? We made some improvements to the esbuild binary resolution and I'm hoping it might help here as well.

**Comment by zaiste** [comment]
@BRKalow it works now **without** adding `esbuild` explicitly. Thank you!
